x-pack/plugins/infra/public/containers/logs/log_position/log_position_state.ts

```diff
@@ -79,26 +119,88 @@ export const useLogPositionState: () => LogPositionStateParams & LogPositionCall
     [startKey, endKey]
   );
 
+  // Allow setting `startDate` and `endDate` separately, or together
+  const updateDateRange = useCallback(
+    (newDateRange: Partial<DateRange>) => {
+      // Prevent unnecessary re-renders
+      if (!('startDateExpression' in newDateRange) && !('endDateExpression' in newDateRange)) {
+        return;
+      }
+
+      const nextStartDateExpression =
+        newDateRange.startDateExpression || dateRange.startDateExpression;
+      const nextEndDateExpression = newDateRange.endDateExpression || dateRange.endDateExpression;
+
+      if (!isValidDatemath(nextStartDateExpression) || !isValidDatemath(nextEndDateExpression)) {
+        return;
+      }
+
+      // Dates are valid, so the function cannot return `null`
+      const nextStartTimestamp = datemathToEpochMillis(nextStartDateExpression)!;
+      const nextEndTimestamp = datemathToEpochMillis(nextEndDateExpression, 'up')!;
+
+      // Reset the target position if it doesn't fall within the new range.
+      if (
+        targetPosition &&
+        (nextStartTimestamp > targetPosition.time || nextEndTimestamp < targetPosition.time)
+      ) {
+        jumpToTargetPosition(null);
+      }
+
+      setDateRange({
+        ...newDateRange,
+        startTimestamp: nextStartTimestamp,
+        endTimestamp: nextEndTimestamp,
+        timestampsLastUpdate: Date.now(),
+      });
+    },
+    [setDateRange, dateRange, targetPosition]
+  );
+
+  // `endTimestamp` update conditions
+  useEffect(() => {
+    if (dateRange.endDateExpression !== 'now') {
+      return;
+    }
+
+    // User is close to the bottom edge of the scroll. The value of
+    // `pagesAfterEnd` changes on live stream as well.
```

Comment:
This doesn't seem to be universally true.

Reply:
I let it running for me and it works, sometimes, sometimes it doesn't 🤷‍♂.

What _always_ changes for sure is the `visiblePositions` object that holds `pagesAfterEnd`. When I access the property through the object (and use the object as a dependency) it works correctly.